### ISSUE TITLE
Telegram Integration

### ISSUE THREAD

**Issue by wagnst** [zammad/zammad#362]
Please integrate a Telegram Support-Chat Integration

See: https://telegram.org/

**Comment by martini** [comment]
I'm just reading the telegram api. How could an implementation look like?

Would it be a TelegramBot? Any feedback would be welcome!

**Comment by wagnst** [comment]
Yes, I could imagine integrating with a Bot would be a good option. A cusomter writes the bot, which connects to the zammand backend and creates a ticket there. 

Alternatively you could use the `telegram-cli`, which represents a fully working cli client for telegram.

Advantages are, that youre not bound to a phone (you can use the messenger in parallel on multiple devices) and event without a mobile number - and of course all open source, end to end encrypted and the API ;-)

**Comment by dynamicman36** [comment]
+1

**Comment by schurig** [comment]
+1

**Comment by schurig** [comment]
This seems to be what we are looking for: https://fullmeter.com/blog/?p=14

Downside: ssl cert required. But I suppose that everyone whose zammad is public accessible is using a ssl certificate

**Comment by schurig** [comment]
How would a public webhook endpoint look like for a new Telegram Channel?

something like `/api/v1/telegram`?

(maybe @martini, @mrflix, @monotek, @thorsteneckel can help me?)

**Comment by martini** [comment]
Big thanks to @schurig for implementing Telegram for Zammad! 🍾 It's available in Zammad 1.3 -> https://zammad.com/news/release-1-3-batch-processing_telegram_integration

**Comment by chrisdecker1201** [comment]
First nice work. Thanks a lot. I just configured it and all works fine, but when I try to answer a opened ticket in zammad, the message never gets out. First I thought maybe it's because it's my bot and I'm writing to it with the telegram account which relies on it, but I checked it with a mate and it's the same. I'm glade to help with debugging or maybe I just missed a little thing.

**Comment by thorsteneckel** [comment]
Hi @chrisdecker1201 - could you please create a new issue for this and provide the information requested in the issue template? The log file would be very interesting here. Please make sure that the scheduler works correctly. Otherwise the messages won't get send.
